lib/PasswordStrategy.js

```diff
@@ -48,7 +48,8 @@ function Strategy(options) {
             sysPassword: password
           }, function(err, verified) {
             if(err) {
-              return callback(err);
+              // FIXME: Ignore errors to ensure all IDs are checked
+              return callback();
```

Comment:
> This doesn't seem right. An error should not be thrown when a password is invalid, it should result in verified being false -- is that not the case here? What error was being thrown?

An error isn't thrown if a password is invalid, verified is set to false. However, we have 3 cases where errors are thrown, and in each case, if we return callback(err), then we won't check the other identities which may provide the proper match. To put it another way, every error that can be thrown by the called code should be ignored because a further check may find an identity that does pass/match.

We're going to have to walk through this in person as what you've stated above seems like the wrong thing to do when using async.each() and the types of err values being returned. At present, every error value should be ignored until you check all of the identities against the provided password. If you don't do that, you can't say whether or not the given password worked.